### PR TITLE
[jit] Fix a constant propagation problem on 32 bit with iconst+conv.i8, we would be setting inst_imm even for long opcodes.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins backport 2018-06

**Comment by vargaz** [comment]
This fixes a mini test failure on 32 bit ios + llvm.

File: mono/mini/local-propagation.c [hunk 1]
```diff
@@ -669,11 +669,13 @@ mono_local_cprop (MonoCompile *cfg)
 					opcode2 = mono_op_to_op_imm (ins->opcode);
 					if ((opcode2 != -1) && mono_arch_is_inst_imm (ins->opcode, opcode2, def->inst_c0) && ((srcindex == 1) || (ins->sreg2 == -1))) {
 						ins->opcode = opcode2;
-						if ((def->opcode == OP_I8CONST) && (sizeof (gpointer) == 4)) {
+						if ((def->opcode == OP_I8CONST) && (sizeof (gpointer) == 4))
 							ins->inst_l = def->inst_l;
-						} else {
+						else if (regtype == 'l' && sizeof (gpointer) == 4)
```
Comment (kumpera) [inline mono/mini/local-propagation.c]: This should use a TARGET_SIZEOF_ define.

**Comment by marek-safar** [comment]
@monojenkins backport 2018-08

**Comment by luhenry** [comment]
@vargaz what's the status of this PR?

**Comment by luhenry** [comment]
@monojenkins build failed

**Comment by alexischr** [comment]
Failures on Windows i386 are https://github.com/mono/mono/issues/10508

**Comment by vargaz** [comment]
https://github.com/mono/mono/pull/10750 fixes it.

**Comment by lewurm** [comment]
@monojenkins backport 2018-08

previous backport (https://github.com/mono/mono/pull/10326) was closed.

**Comment by lewurm** [comment]
@monojenkins backport 2018-08

**Pull request merged**